src/box/txn_limbo.c

```diff
@@ -530,8 +594,8 @@ txn_limbo_read_confirm(struct txn_limbo *limbo, int64_t lsn)
 static void
 txn_limbo_confirm_lsn(struct txn_limbo *limbo, int64_t confirm_lsn)
 {
-	txn_limbo_write_confirm(limbo, confirm_lsn);
-	txn_limbo_read_confirm(limbo, confirm_lsn);
+	limbo->volatile_confirmed_lsn = confirm_lsn;
```

Comment:
Lets add here and in other volatile LSN updates an assertion that the new LSN is > the old one.